test/burn/org/apache/cassandra/utils/memory/LongBufferPoolTest.java

```diff
@@ -327,24 +371,35 @@ BufferCheck sample()
 
                     return checks.get(index);
                 }
-
-                private int sum1toN(int n)
-                {
-                    return (n * (n + 1)) / 2;
-                }
             }));
         }
 
-        boolean first = true;
         while (!latch.await(10L, TimeUnit.SECONDS))
         {
-            if (!first)
-                BufferPool.assertAllRecycled();
-            first = false;
+            int stalledThreads = 0;
+            int doneThreads = 0;
+
             for (AtomicBoolean progress : makingProgress)
             {
-                assert progress.get();
-                progress.set(false);
+                if (progress.getAndSet(false) == false)
```

Comment:
```suggestion
                if (!progress.getAndSet(false))
```

Reply:
I left the assert alone as on 346, seems like there shouldn't be any side-effects from executing an assert (and they're optional so it could affect correctness if somebody ran without -ea). The getAndSet isn't required for correctness as the worker thread only changes from false -> true and doesn't take any action based on the flag, and it will fix it in the next cycle.